source/data-processing.rst

```diff
@@ -0,0 +1,53 @@
+.. _doc_routinator_data_processing:
+
+Data Processing
+===============
+
+Fetching
+--------
+
+There are two protocols in use to transport RPKI data: rsync and the RPKI
+Repository Delta Protocol (RRDP), which relies on HTTPS. RRDP was designed to be
+the successor to rsync. As almost all RPKI repositories currently support both
+protocols, Routinator will prefer RRDP if available. 
+
+When traversing the RPKI tree, Routinator will find several pointers to child
+publication points, such as the ones operated by National Internet Registries
+and organisations running delegated RPKI. Each pointer explicitly states if RRDP
+is supported in addition to rsync. 
+
+If an RRDP endpoint is unavailable but it has worked in the past, Routinator
+will assume this is a transient problem. It will retry using RRDP for 60 minutes
+since the last successful update, during which it will rely on the locally
+cached data for this repository. After this time, Routinator will try to use
+rsync to fetch the data instead. If this is unsuccessful too, the local cache is
+used until the objects go stale and ultimately expire. The fallback time can be
+changed using the :option:`--rrdp-fallback-time` option.
+
+Routinator will fetch new RPKI data ten minutes after the last successful update
+has finished. The interval can be changed using the :option:`--refresh` option.
+It is possible that temporary network problems cause a repository to be very
+slow to update. To ensure a slow repository doesn't stop the entire update
+process from completing, Routinator has a timeout of five minutes for stalled
```

Comment:
What exactly is this a timeout on? An entire single attempt to update from the repository (i.e. including DNS resolution, initial connection, time to first byte, all the way up to the last byte, or only part of this process)? Does it also include retries, i.e. if 2 minutes are spent updating but that fails in some way and a retry is attempted does the 5 minute timer restart or are 2 minutes of it used already?

Reply:
@partim Can you clarify this and is there a difference between rsync and RRDP?

Reply:
For rsync, the timeout is around the spawned rsync process, i.e., around everything. For RRDP, the timeout is actually an HTTP request timeout. So it’s 5 minutes for the notification file and then five minutes for the snapshot or five minutes for each delta.

Reply:
Note that these timeouts are individually tweakable via the `rsync-timeout` and `rrdp-timeout` options.